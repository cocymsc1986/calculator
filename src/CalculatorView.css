.calculator {
  background: black;
  padding: 1rem;
  display: flex;
  flex-flow: wrap;
  align-content: space-between;
  width: 100%;
  max-width: 700px;
  margin: 0 auto;
}

.calculator__screen {
  background: #ffffff;
  width: 100%;
  font-size: 4rem;
  text-align: right;
  padding: 0.5rem 1rem;
  margin-bottom: 1rem;
}

.calculator__button {
  width: 25%;
  font-size: 2rem;
  padding: 1rem 0;
}
