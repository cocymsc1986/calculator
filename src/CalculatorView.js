import React from "react";

export default ({ onRegularClick, onClearClick, onCEClick, onSumClick }) => (
  <div>Calculator</div>
);
