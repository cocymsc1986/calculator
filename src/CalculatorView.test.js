import React from "react";
import { shallow } from "enzyme";

import CalculatorView from "./CalculatorView";

describe("CalculatorView", () => {
  it("should render", () => {});

  it("should call call onRegularClick when clicking a standard button", () => {});

  it("should call call onClearClick when clicking clear button", () => {});

  it("should call call onCEClick when clicking CE button", () => {});

  it("should call call onSumClick when clicking equals button", () => {});
});
