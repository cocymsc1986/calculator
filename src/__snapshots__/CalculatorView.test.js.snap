// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorView should render 1`] = `
<div
  className="calculator"
>
  <div
    className="calculator__screen-container"
  >
    <div
      className="calculator__screen"
    />
    <img
      alt="logo"
      className="calculator__logo"
      src="logo.svg"
    />
  </div>
  <div
    className="calculator__body"
  >
    <div
      className="calculator__button"
    >
      <button
        data-test-id="clear"
      >
        C
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="ce"
      >
        CE
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="modulus"
      >
        %
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="divide"
        value="/"
      >
        /
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-1"
        value="1"
      >
        1
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-2"
        value="2"
      >
        2
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-3"
        value="3"
      >
        3
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="multiply"
        value="*"
      >
        x
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-4"
        value="4"
      >
        4
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-5"
        value="5"
      >
        5
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-6"
        value="6"
      >
        6
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="subtract"
        value="-"
      >
        -
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-7"
        value="7"
      >
        7
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-8"
        value="8"
      >
        8
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-9"
        value="9"
      >
        9
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="add"
        value="+"
      >
        +
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="decimal"
        value="."
      >
        .
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="number-0"
        value="0"
      >
        0
      </button>
    </div>
    <div
      className="calculator__button"
    >
      <button
        data-test-id="equals"
        value="="
      >
        =
      </button>
    </div>
    <div
      className="calculator__button"
    />
  </div>
</div>
`;
